R/setkey.R

```diff
@@ -332,6 +332,7 @@ CJ <- function(..., sorted = TRUE, unique = FALSE)
     dups = FALSE # fix for #1513
     # using rep.int instead of rep speeds things up considerably (but attributes are dropped).
     j = lapply(l, class)  # changed "vapply" to avoid errors with "ordered" "factor" input
+    tzones = lapply(l, function(col) attr(col, 'tzone'))
```

Comment:
This can be  tzones = lapply(l, attr, 'tzone')  please and can it be moved down inside the `if()` so that it is only calculated when needed?